core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -76,66 +86,66 @@
     private static final PositionsAppenderFactory POSITIONS_APPENDER_FACTORY = new PositionsAppenderFactory(new BlockTypeOperators());
 
     @Test(dataProvider = "types")
-    public void testMixedBlockTypes(Type type)
+    public void testMixedBlockTypes(Type type, Function<Block, Block> blockAdaptation)
```

Comment:
Introduce composite `TestType` that composes of `Type type` and `Function<Block, Block> blockAdaptation`
Then you wouldn't have to pass `blockAdaptation` around and it will be more explicit what is tested, e.g:
```
VARCHAR,
VARCHAR_WITH_TEST_BLOCK
```
`TestType` could have default constructor so that you wouldn't have to use `identity` arg by default

Reply:
done